Marlin/src/gcode/feature/runout/M412.cpp

```diff
@@ -31,22 +31,28 @@
  * M412: Enable / Disable filament runout detection
  */
 void GcodeSuite::M412() {
-  if (parser.seen("HS"
+  if (parser.seen("HSD"
     #if ENABLED(HOST_ACTION_COMMANDS)
       "R"
     #endif
   )) {
     #if ENABLED(HOST_ACTION_COMMANDS)
       if (parser.seen('H')) runout.host_handling = parser.value_bool();
     #endif
-    const bool seenR = parser.seen('R'), seenS = parser.seen('S');
+    const bool seenR = parser.seen('R'), seenS = parser.seen('S'), seenD = parser.seen('D');
     if (seenR || seenS) runout.reset();
     if (seenS) runout.enabled = parser.value_bool();
+    #ifdef FILAMENT_RUNOUT_DISTANCE_MM
+      if (seenD) RunoutResponseDelayed::runout_distance_mm = parser.value_linear_units();
+    #endif
```

Comment:
This cannot work! You must call `parser.value...` immediately after `seen` for the parameter word whose value you want. The reason `R` and `S` are okay to combine above is that they are mutually-exclusive.

Reply:
Fixed by moving `seen('D')` into the `if` clause.